woocommerce/woocommerce-android
# Business logic updates for IPP feedback survey banner

#### Business logic updates for IPP feedback survey banner

**Note:** This PR is part of the IPP in-app feedback banner project https://github.com/woocommerce/woocommerce-android/issues/8116. It targets the feature branch feature/8116-ipp-feedback-banner.

### Description


This PR introduces the following new use case classes:
* `MarkFeedbackBannerAsDismissed`
* `MarkFeedbackBannerAsDismissedForever`
* `MarkIPPFeedbackSurveyAsCompleted`

that are responsible for storing banner actions in `SharedPreferences`.

It also adds the following updates to existing use cases:
* `GetIPPFeedbackBannerData` - renamed from `GetIPPFeedbackSurveyUrl` and modified to return banner message along with survey url. These values are wrapped up by `IPPFeedbackBanner ` data class.
* `ShouldShowFeedbackBanner` - updated to limit showing the banner only to stores in CA and US.

### Testing instructions


There is no UI part associated with the use cases yet, so I just recommend reviewing unit tests: `ShouldShowFeedbackBannerTest` and `GetIPPFeedbackBannerDataTest` and make sure they are complete.

### Roadmap
In the next PR, I'll add a feature flag and UI for the banner.

### Images/gif
